LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -1061,5 +1961,118 @@ private static StructProperty GenerateExpressionGUID()
+            }
+            return -1;
+        }
+
+        /// <summary>
+        /// Gets a union of two strings. Assumes both only difer at the end.
+        /// </summary>
+        /// <param name="s1">First string.</param>
+        /// <param name="s2">Second string.</param>
+        /// <returns>Union of input strings.</returns>
+        private static string GetUnionOfStrings(string s1, string s2)
+        {
+            // Get the old name found in all pieces of the conversation by getting the union of the bioconversation
+            // and the wwise bank names.
+            // Assumes that both begin the same, since that's the behavior in all vanilla occurences, and helps
+            // keep the logic simple.
+            string union = "";
+            for (int w = 0, b = 0; w < s1.Length && b < s2.Length; w++, b++)
+            {
+                if (char.ToLower(s1[w]) == char.ToLower(s2[b]))
+                {
+                    union += char.ToLower(s1[w]);
+                }
+                else
+                {
+                    break;
+                }
+            }
+            return union;
+        }
+
+        /// <summary>
+        /// Generates a FNV132 hash of the given name.
+        /// IMPORTANT: This may not be compeletely bug-free or may be missing a couple of details, but so far it works.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>The decimal representation of the hash.</returns>
+        private static uint GetBankId(string name)
+        {
+            byte[] bytedName = Encoding.ASCII.GetBytes(name.ToLower()); // Wwise automatically lowecases the input
+
+            // FNV132 hashing algorithm
+            uint hash = 2166136261;
+            foreach (byte namebyte in bytedName)
+            {
+                hash = ((hash * 16777619) ^ namebyte) & 0xFFFFFFFF;
+            }
+            return hash;
+        }
+
+        /// <summary>
+        /// Convert a big endian hex to its little endian representation.
+        /// </summary>
+        /// <param name="bigEndian">Endian to convert.</param>
+        /// <returns>Little endian.</returns>
+        private static string BigToLittleEndian(string bigEndian)
+        {
+            byte[] asCurrentEndian = new byte[4];
+            string littleEndian = "";
+            for (int i = 0; i < 4; i++)
+            {
+                asCurrentEndian[i] = Convert.ToByte(bigEndian.Substring(i * 2, 2), 16);
+                littleEndian = $"{asCurrentEndian[i]:X2}{littleEndian}";
+            }
+            return littleEndian;
+        }
+
+        /// <summary>
+        /// Create a var link with custom properties.
+        /// </summary>
+        /// <param name="pew">Current PE window.</param>
+        /// <param name="name">LinkDesc.</param>
+        /// <returns>The varLink StructProperty.</returns>
+        private static StructProperty CreateVarLink(PackageEditorWindow pew, string name)
```

Comment:
This function does not need a PackageEditorWindow, it needs an IMEPackage. The parameters should reflect that.

Reply:
True